test/test_http11.py

```diff
@@ -329,6 +329,23 @@ def test_chunked_overrides_body(self):
 
         assert received == expected
 
+    def test_response_with_empty_reason(self):
+        c = HTTP11Connection('httpbin.org')
+        c._sock = sock = DummySocket()
+
+        sock._buffer = BytesIO(
+            b"HTTP/1.1 201\r\n"
```

Comment:
Interestingly, *this* is not a valid response. You're right that the reason phrase can be absent, but the space in front of it cannot. So this should be `b"HTTP/1.1 201 \r\n"` instead.